aws/src/main/java/org/apache/iceberg/aws/s3/S3InputStream.java

```diff
@@ -57,6 +64,14 @@ class S3InputStream extends SeekableInputStream implements RangeReadable {
   private final Counter readOperations;
 
   private int skipSize = 1024 * 1024;
+  private RetryPolicy<Object> retryPolicy =
+      RetryPolicy.builder()
+          .handle(
+              ImmutableList.of(
+                  SSLException.class, SocketTimeoutException.class, SocketException.class))
```

Comment:
Should `software.amazon.awssdk.core.exception.SdkClientException` be included in the exception list ?
It  indicates issues with the client-side networking stack, such as network timeouts.

Reply:
Judging by the comment in the SdkClientException class, it should not be retryable as there might be multiple reasons for it 

```
Base type for all client exceptions thrown by the SDK. This exception is thrown when service could not be contacted for a response, or when client is unable to parse the response from service.
Exceptions that extend SdkClientException are assumed to be not retryable, with a few exceptions:
RetryableException - usable when calls should explicitly be retried
Exceptions mentioned as a retryable exception in SdkDefaultRetrySetting
See Also:
SdkServiceException
```

Is there a more specific one that you were thinking?

Reply:
Looking at https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/core/internal/retry/SdkDefaultRetrySetting.html, we can add `RETRYABLE_EXCEPTIONS` to the list.